### ISSUE TITLE
Galera fails to build on Debian/hurd-i386 due to missing platform definition in chromium/build_config.h

### ISSUE THREAD

**Issue by ottok** [codership/galera#320]
Log at https://buildd.debian.org/status/fetch.php?pkg=galera-3&arch=hurd-i386&ver=25.3.10-1&stamp=1438198070 fails with:

```
Checking for C library check... yes
Checking for C library m... yes
Checking for C library rt... yes
 -std=c99 -fno-strict-aliasing -pipe
 -g -O3 -DNDEBUG -m32 -march=i686 -Wall -Wextra -Wno-unused-parameter 
-D_FORTIFY_SOURCE=2 -pthread -D_XOPEN_SOURCE=600 -DHAVE_COMMON_H -DHAVE_BYTESWAP_H -DHAVE_ENDIAN_H -DHAVE_BOOST_SHARED_PTR_HPP -DHAVE_TR1_UNORDERED_MAP -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG=1 -DHAVE_ASIO_HPP -DHAVE_ASIO_SSL_HPP -Werror -DGCS_USE_GCOMM -DGALERA_LOG_H_ENABLE_CXX -Wno-variadic-macros -Wno-old-style-cast -Wno-missing-field-initializers -Wno-effc++
 -Wno-long-long -Wno-deprecated -ansi -pipe -Weffc++ -Wold-style-cast
scons: done reading SConscript files.
scons: Building targets ...
g++ -o galera/src/certification.o -c -Wno-long-long -Wno-deprecated -ansi -pipe -Weffc++ -Wold-style-cast -g -O3 -DNDEBUG -m32 -march=i686 -Wall -Wextra -Wno-unused-parameter -pedantic -D_FORTIFY_SOURCE=2 -pthread -D_XOPEN_SOURCE=600 -DHAVE_COMMON_H -DHAVE_BYTESWAP_H -DHAVE_ENDIAN_H -DHAVE_BOOST_SHARED_PTR_HPP -DHAVE_TR1_UNORDERED_MAP -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG=1 -DHAVE_ASIO_HPP -DHAVE_ASIO_SSL_HPP -Werror -I. -Iasio -Icommon -Igalerautils/src -Igcomm/src -Igcomm/src/gcomm -Igcache/src -Igcs/src -Iwsdb/src -Igalera/src galera/src/certification.cpp
In file included from ./chromium/compiler_specific.h:8:0,
                 from ./chromium/aligned_memory.h:39,
                 from galerautils/src/gu_reserved_container.hpp:19,
                 from galerautils/src/gu_vector.hpp:25,
                 from galerautils/src/gu_rset.hpp:16,
                 from galera/src/key_set.hpp:9,
                 from galera/src/write_set_ng.hpp:18,
                 from galera/src/trx_handle.hpp:14,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
./chromium/build_config.h:50:2: error: #error Please add support for your platform in build/build_config.h
 #error Please add support for your platform in build/build_config.h
  ^
./chromium/build_config.h:148:2: error: #error Please add support for your compiler in build/build_config.h
 #error Please add support for your compiler in build/build_config.h
  ^
In file included from galera/src/write_set.hpp:9:0,
                 from galera/src/trx_handle.hpp:9,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
galera/src/key_os.hpp: In member function 'size_t galera::KeyOS::unserialize(const byte_t*, size_t, size_t)':
galera/src/key_os.hpp:308:5: error: control reaches end of non-void function [-Werror=return-type]
     }
     ^
cc1plus: all warnings being treated as errors
scons: *** [galera/src/certification.o] Error 1
scons: building terminated because of errors.
```

Indeed, the file chromium/build_config.h does not define anything about platform hurd.

**Comment by ottok** [comment]
This applies also for debian/kfreebsd-i386 and -amd64:
https://buildd.debian.org/status/fetch.php?pkg=galera-3&arch=kfreebsd-i386&ver=25.3.10-1&stamp=1438105254
https://buildd.debian.org/status/fetch.php?pkg=galera-3&arch=kfreebsd-amd64&ver=25.3.10-1&stamp=1438259906

```
Checking for C library pthread... yes
Checking for C library rt... yes
Checking for C header file sys/epoll.h... no
Checking for C header file byteswap.h... yes
Checking for C header file endian.h... yes
Checking for C++ header file boost/shared_ptr.hpp... yes
Checking for C++ header file unordered_map... no
Checking for C++ header file tr1/unordered_map... yes
Checking for C++ library boost_program_options... yes
Checking for C++ header file asio.hpp... yes
Checking for C++ header file asio/ssl.hpp... yes
Checking for C library ssl... yes
Checking for C library crypto... yes
Checking for C header file check.h... yes
Checking for C library check... yes
Checking for C library m... yes
Checking for C library rt... yes
 -std=c99 -fno-strict-aliasing -pipe
 -g -O3 -DNDEBUG -m64 -Wall -Wextra -Wno-unused-parameter 
-D_FORTIFY_SOURCE=2 -pthread -D_XOPEN_SOURCE=600 -DHAVE_COMMON_H -DHAVE_BYTESWAP_H -DHAVE_ENDIAN_H -DHAVE_BOOST_SHARED_PTR_HPP -DHAVE_TR1_UNORDERED_MAP -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG=1 -DHAVE_ASIO_HPP -DHAVE_ASIO_SSL_HPP -Werror -DGCS_USE_GCOMM -DGALERA_LOG_H_ENABLE_CXX -Wno-variadic-macros -Wno-old-style-cast -Wno-missing-field-initializers -Wno-effc++
 -Wno-long-long -Wno-deprecated -ansi -pipe -Weffc++ -Wold-style-cast
scons: done reading SConscript files.
scons: Building targets ...
g++ -o galera/src/certification.o -c -Wno-long-long -Wno-deprecated -ansi -pipe -Weffc++ -Wold-style-cast -g -O3 -DNDEBUG -m64 -Wall -Wextra -Wno-unused-parameter -pedantic -D_FORTIFY_SOURCE=2 -pthread -D_XOPEN_SOURCE=600 -DHAVE_COMMON_H -DHAVE_BYTESWAP_H -DHAVE_ENDIAN_H -DHAVE_BOOST_SHARED_PTR_HPP -DHAVE_TR1_UNORDERED_MAP -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG=1 -DHAVE_ASIO_HPP -DHAVE_ASIO_SSL_HPP -Werror -I. -Iasio -Icommon -Igalerautils/src -Igcomm/src -Igcomm/src/gcomm -Igcache/src -Igcs/src -Iwsdb/src -Igalera/src galera/src/certification.cpp
g++ -o galera/src/certification.os -c -Wno-long-long -Wno-deprecated -ansi -pipe -Weffc++ -Wold-style-cast -g -O3 -DNDEBUG -m64 -Wall -Wextra -Wno-unused-parameter -pedantic -fPIC -D_FORTIFY_SOURCE=2 -pthread -D_XOPEN_SOURCE=600 -DHAVE_COMMON_H -DHAVE_BYTESWAP_H -DHAVE_ENDIAN_H -DHAVE_BOOST_SHARED_PTR_HPP -DHAVE_TR1_UNORDERED_MAP -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG=1 -DHAVE_ASIO_HPP -DHAVE_ASIO_SSL_HPP -Werror -I. -Iasio -Icommon -Igalerautils/src -Igcomm/src -Igcomm/src/gcomm -Igcache/src -Igcs/src -Iwsdb/src -Igalera/src galera/src/certification.cpp
In file included from ./chromium/compiler_specific.h:8:0,
                 from ./chromium/aligned_memory.h:39,
                 from galerautils/src/gu_reserved_container.hpp:19,
                 from galerautils/src/gu_vector.hpp:25,
                 from galerautils/src/gu_rset.hpp:16,
                 from galera/src/key_set.hpp:9,
                 from galera/src/write_set_ng.hpp:18,
                 from galera/src/trx_handle.hpp:14,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
./chromium/build_config.h:50:2: error: #error Please add support for your platform in build/build_config.h
 #error Please add support for your platform in build/build_config.h
  ^
./chromium/build_config.h:148:2: error: #error Please add support for your compiler in build/build_config.h
 #error Please add support for your compiler in build/build_config.h
  ^
In file included from ./chromium/compiler_specific.h:8:0,
                 from ./chromium/aligned_memory.h:39,
                 from galerautils/src/gu_reserved_container.hpp:19,
                 from galerautils/src/gu_vector.hpp:25,
                 from galerautils/src/gu_rset.hpp:16,
                 from galera/src/key_set.hpp:9,
                 from galera/src/write_set_ng.hpp:18,
                 from galera/src/trx_handle.hpp:14,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
./chromium/build_config.h:50:2: error: #error Please add support for your platform in build/build_config.h
 #error Please add support for your platform in build/build_config.h
  ^
./chromium/build_config.h:148:2: error: #error Please add support for your compiler in build/build_config.h
 #error Please add support for your compiler in build/build_config.h
  ^
In file included from galera/src/write_set.hpp:9:0,
                 from galera/src/trx_handle.hpp:9,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
galera/src/key_os.hpp: In member function 'size_t galera::KeyOS::unserialize(const byte_t*, size_t, size_t)':
galera/src/key_os.hpp:308:5: error: control reaches end of non-void function [-Werror=return-type]
     }
     ^
In file included from galera/src/write_set.hpp:9:0,
                 from galera/src/trx_handle.hpp:9,
                 from galera/src/certification.hpp:8,
                 from galera/src/certification.cpp:5:
galera/src/key_os.hpp: In member function 'size_t galera::KeyOS::unserialize(const byte_t*, size_t, size_t)':
galera/src/key_os.hpp:308:5: error: control reaches end of non-void function [-Werror=return-type]
     }
     ^
cc1plus: all warnings being treated as errors
scons: *** [galera/src/certification.o] Error 1
cc1plus: all warnings being treated as errors
scons: *** [galera/src/certification.os] Error 1
scons: building terminated because of errors.
```

**Comment by ayurchen** [comment]
Otto, how important is it to compile  for these platforms? What is the proper/best way to test it (are there any install images? public machines?)

**Comment by ottok** [comment]
I have access to some of the platforms, so if you provide me some patches I can try how they work. I understand however that it will be difficult for you to develop fixes for these without access to alternative platforms. Some might be available as virtual machines for anybody, at least hurd and kfreebsd are just basic Debian machines but with a non-Linux kernel.

Hurd and kfreebsd are tough not the highest priorities. Please fix #319 and #321 first.

**Comment by ottok** [comment]
Hurd-i386 does still not build fully, but the build_config.h issue is fixed, so closing this issue now.
